### PR TITLE
AMB-21 Created client DAO

### PR DESCRIPTION
**Pull request by dana-kainos**

### CODE REVIEW

File: src/main/java/org/soniakbew/daos/ClientDao.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package org.soniakbew.daos;
+
+import org.soniakbew.models.Client;
+
```
Comment (lukemKainos) [inline src/main/java/org/soniakbew/daos/ClientDao.java]: space?

File: src/main/java/org/soniakbew/daos/ClientDao.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            Statement statement = connection.createStatement();
+
+            ResultSet resultSet = statement.executeQuery(
+                    "SELECT clientId, name FROM `client`;"
+            );
+
+            while (resultSet.next()) {
+                Client client = new Client(
+                        resultSet.getInt("clientId"),
+                        resultSet.getString("name")
```
Comment (lukemKainos) [inline src/main/java/org/soniakbew/daos/ClientDao.java]: consider abstracting this out to a mapper for both gets

File: src/main/java/org/soniakbew/daos/ClientDao.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return new Client(
+                        resultSet.getInt("clientId"),
+                        resultSet.getString("name")
+                );
+            }
+        }
+        return null;
```
Comment (lukemKainos) [inline src/main/java/org/soniakbew/daos/ClientDao.java:51]: what happens when this returns null, does it get caught further up or return null to the client?

**Review by lukemKainos — CHANGES_REQUESTED** [review]

File: src/main/java/org/soniakbew/daos/ClientDao.java (hunk 3 above)
Comment (dana-kainos) [reply src/main/java/org/soniakbew/daos/ClientDao.java:51]: gets caught further up, in the service class. would this be bad practice?

File: src/main/java/org/soniakbew/daos/ClientDao.java (hunk 2 above)
Comment (dana-kainos) [reply src/main/java/org/soniakbew/daos/ClientDao.java]: will be abstracted out to a mapper in the service class

File: src/main/java/org/soniakbew/daos/ClientDao.java (hunk 3 above)
Comment (lukemKainos) [reply src/main/java/org/soniakbew/daos/ClientDao.java:51]: no, just so long as i know its present thats fine

File: src/main/java/org/soniakbew/daos/ClientDao.java (hunk 2 above)
Comment (lukemKainos) [reply src/main/java/org/soniakbew/daos/ClientDao.java]: if its abstracted in the service class why is it in here?

**Review by lukemKainos — APPROVED** [review]

**Review by james-liam-butler — APPROVED** [review]

**Pull request merged**